nntrainer/tensor/cache_pool.cpp

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file   cache_pool.cpp
+ * @date   01 July 2022
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Cache pool class inherited from memory pool
+ *
+ */
+
+#include <limits>
+#include <numeric>
+#include <vector>
+
+#include <cache_pool.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <profiler.h>
+
+namespace nntrainer {
+
+void CacheElem::swapIn() {
+  std::lock_guard<std::mutex> lock(device_mutex);
+  void *buf = device->getBuffer(offset, length);
+  mem_data->setAddr((float *)buf);
+  mem_data->setValid(true);
+  active = true;
+
+  std::string msg("CacheElem(");
+  msg += device->getDevicePath() + ") #" + std::to_string(id);
+  PROFILE_MEM_ALLOC(buf, length, msg);
+}
+
+void CacheElem::swapOut() {
+  std::lock_guard<std::mutex> lock(device_mutex);
+  void *buf = (void *)mem_data->getAddr();
+  device->putBuffer(buf);
+  mem_data->setValid(false);
+  active = false;
+
+  PROFILE_MEM_DEALLOC(buf);
+}
+
+CachePool::CachePool(const std::string &name) :
+  swap_device(std::make_shared<SwapDevice>(name + std::to_string(getpid()))) {
```

Comment:
I think the swap_device is needs to be set for every ```CachePool``` and I think we might needs multiple memory pool in later usage scenario. So.. setting the swap device name should not be set by using just getpid().
It might be considered in later PR though.

Reply:
Multiple memory pool case is not considered. I'll think of it more,
Thanks for your advice.